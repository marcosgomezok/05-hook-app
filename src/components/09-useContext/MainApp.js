import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext';
import { useState } from 'react';

export const MainApp = () => {

   const [user, setUser] = useState({})

    return (
            <UserContext.Provider value={{
                user,setUser
            }}>

                <AppRouter/>

            </UserContext.Provider>
    )
}
