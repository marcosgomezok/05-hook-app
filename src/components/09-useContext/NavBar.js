import React from 'react'
//import {Link} from 'react-router-dom';
import {NavLink} from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

            <NavLink to="/" className="navbar-brand" >useContext</NavLink>

            <div className="collapse navbar-collapse" id="navbarNav">

                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" to="/" className="nav-item nav-link" >Home</NavLink>
                    <NavLink exact activeClassName="active" to="/about" className="nav-item nav-link" >About</NavLink>
                    <NavLink exact activeClassName="active" to="/login" className="nav-item nav-link" >Login</NavLink>
                </div>                    
            </div>
        </nav>
    )
}

/* <nav>
            <ul>
                <li><Link to="./">Home</Link></li>
                <li><Link to="./about">About</Link></li>
                <li><Link to="./login">Login</Link></li>
            </ul>
        </nav> */