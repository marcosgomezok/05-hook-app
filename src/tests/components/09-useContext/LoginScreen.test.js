
import  React from 'react';
import { shallow } from 'enzyme';
import { LoginScreen } from './../../../components/09-useContext/LoginScreen';
import { UserContext } from './../../../components/09-useContext/UserContext';


describe('Pruebas en <LoginScreen/>', () => {
    
    const setUser = jest.fn();
    //despues cambiar shallow por mount
    const wrapper = shallow(<UserContext.Provider value={{setUser}}>

        <LoginScreen/>

    </UserContext.Provider>)
    
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de ejecutar el setUser con el argumento esperado ', () => {
        //hacer funcionar el mount primero
        /* wrapper.find('button').prop('onClick')();

        expect(setUser).toHaveBeenCalledWith({
            id:123,
            name: 'Marcos'
        }); */
    })
    
})
