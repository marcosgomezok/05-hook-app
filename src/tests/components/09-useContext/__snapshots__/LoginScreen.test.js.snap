// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe de mostrarse correctamente 1`] = `<LoginScreen />`;
